pkgs/applications/office/arianna/default.nix

```
@@ -0,0 +1,67 @@
+{ lib
+, mkDerivation
+, fetchurl
+, fetchFromGitLab
+
+, cmake
+, extra-cmake-modules
+, reuse
+
+, baloo
+, karchive
+, kconfig
+, kcoreaddons
+, kdbusaddons
+, kfilemetadata
+, ki18n
+, kirigami2
+, kirigami-addons
+, kwindowsystem
+, qqc2-desktop-style
+, qtbase
+, qtquickcontrols2
+, qtwebengine
+, qtwebsockets
+}:
+
+mkDerivation rec {
+  pname = "arianna";
+  version = "1.0.0";
+
+  src = fetchurl {
+    url = "mirror://kde/stable/arianna/arianna-${version}.tar.xz";
+    hash = "sha256-UZnetRo0ozBMCOoIz56qExv92uZ8dyaW5cgI835vARs=";
+  };
```

Why not `fetchgit` from https://invent.kde.org/graphics/arianna

```suggestion
  src = fetchgit {
    url = "https://invent.kde.org/graphics/arianna";
    rev = "v${version}";
    sha256 = "sha256-X3PDGWsQ8Alj5fisZC1tTHQDLPmjtiLw0X9gMvh5KFI=";
  };
```

---

`fetchgit` is generally slower and has more overhead than fetching tarballs. This is one of the main reasons why `fetchFromGitHub` and `fetchFromGitLab` exist (they download tarballs and don't use git).

Is there any reason not to use the well-mirrored official release tarballs?

---

I'm not sure which one is the preferred way really, which is why I was asking :)
I'll leave it to somebody more experienced in NixOS to comment on.